src/common/snippets/src/pass/mlp_seq_tokenization.cpp

```diff
@@ -0,0 +1,207 @@
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "snippets/pass/mlp_seq_tokenization.hpp"
+
+#include "snippets/itt.hpp"
+#include "snippets/op/brgemm.hpp"
+#include "snippets/op/subgraph.hpp"
+#include "snippets/pass/collapse_subgraph.hpp"
+#include "snippets/utils/utils.hpp"
+#include "snippets/utils/tokenization_utils.hpp"
+
+#include "openvino/core/rt_info.hpp"
+#include "openvino/core/validation_util.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "transformations/utils/utils.hpp"
+
+namespace ov {
+namespace snippets {
+namespace pass {
+
+namespace {
+
+inline bool has_one_consumer(const std::shared_ptr<ov::Node>& node) {
+    return node->get_output_target_inputs(0).size() == 1;
+}
+
+size_t get_potential_body_params(const std::shared_ptr<ov::Node>& op) {
+    size_t count = 0;
+    for (size_t i = 1; i < op->get_input_size(); ++i) {
+        const auto input = op->input_value(i);
+        const auto parent = input.get_node_shared_ptr();
+        const auto constant = ov::as_type_ptr<ov::op::v0::Constant>(parent);
+        const auto is_scalar = constant && (ov::shape_size(input.get_shape()) == 1);
+        const auto should_be_inside_body = constant && ov::snippets::op::Subgraph::constant_input_should_be_inside_body(op);
+        const auto is_fq_weight = constant && ov::is_type<ov::op::v0::FakeQuantize>(op);
```

Comment:
Why do we need a separate check on FakeQuantize? From my perspective, it is subject to general rules: if constant is scalar, it is placed inside the body, otherwise - outside. Am I missing something?

Reply:
Aha, so FQ constants contribute to `hidden_virtual_ports_count` instead of `potential_body_params_count`. Actually this separation look like unnecessary complication. Do you think we can combine it in one variable? In this case, we could have all the logic in `get_potential_body_params`

Reply:
I splitted it into two several variables to call `subgraph->set_virtual_port_count(hidden_data_count);` at the end. But then I moved all common tokenization logic into separate helper `tokenize_ordered_nodes(...)` where `hidden_data_count` is calculated and set to tokenized `Subgraph` op. So you're right, now we can combine them into one variable. Thank you for the catch!

@aobolensk, please combine `hidden_virtual_ports_count` and `potential_body_params_count` into `potential_body_params_count`. Thanks in advance!

Reply:
Done